packages/palette-charts/src/elements/DataVis/utils/useHasEnteredViewPort.ts

```diff
@@ -1,24 +1,26 @@
-import { useEffect, useState } from "react"
+import { useEffect, useMemo, useState } from "react"
 
 /**
  * Returns true when component enters viewport
  * @param ref reference to component
  */
 export const useHasEnteredViewport = (ref: React.RefObject<HTMLElement>) => {
+
   const [hasEntered, setHasEntered] = useState(false)
+
+  let observer: IntersectionObserver = null
+
+  if (typeof window !== "undefined"){
```

Comment:
This kind of thing creates problems for server-side rendering because the render will not match when it's hydrated.